File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/glfw/window/hint/OpenGlProfileType.java
package org.saar.lwjgl.glfw.window.hint;

import org.lwjgl.glfw.GLFW;

public enum OpenGlProfileType {

    ANY(GLFW.GLFW_OPENGL_ANY_PROFILE),
    CORE(GLFW.GLFW_OPENGL_CORE_PROFILE),
    COMPATIBILITY(GLFW.GLFW_OPENGL_COMPAT_PROFILE),
    ;

    private final int value;

    OpenGlProfileType(int value) {
        this.value = value;
    }

    public int get() {
        return this.value;
    }
}

File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/glfw/window/OpenGlProfileType.kt
package org.saar.lwjgl.glfw.window

import org.lwjgl.glfw.GLFW

enum class OpenGlProfileType(private val value: Int) {
    ANY(GLFW.GLFW_OPENGL_ANY_PROFILE),
    CORE(GLFW.GLFW_OPENGL_CORE_PROFILE),
    COMPATIBILITY(GLFW.GLFW_OPENGL_COMPAT_PROFILE),
    ;

    fun get() = this.value
}
